### PR TITLE
More inventory interaction enhancements - autofocus-quickclose

### PR DESCRIPTION
**Pull request by OvermindDL1**
After a few others tried the recent change they gave some more ideas for minor QoL improvements.

The first is:  Check if the search box is default focused, and if there's been no other interaction in the window other than just calling for the inventory to close, then do not type in the box but rather defocus it so the inventory then closes.  This is a common use case when opening then closing inventory to just inspect what's there without actually interacting.  This was suggested to me and after testing it I agree that I do instinctively do the same thing, so this makes using the autofocus better.

It does have a user option for this setting, this option is default enabled (it only applies when autofocus is enabled regardless) as it just 'feels right' to multiple that tried it and 'e' being the default key (though I use tab myself hence why I didn't think of this before) is a rare key to start typing with anyway, plus something else can be typed first, but I can set it to default disabled if preferred?

### CODE REVIEW

**Comment by OvermindDL1** [comment]
Also forgot to add the vanilla survival inventory to the autofocus allowlist instead of just NEI's creative interface, fixed here.

I'm thinking I should actually hardcode that one though, thoughts?  Leave it in the file (good example anyway) or hardcode it?  (Or both?)

**Comment by OvermindDL1** [comment]
Also, I'm logging when a GUI match fails for autofocus (when autofocus is on) as it makes it utterly trivial to know what string to copy/paste into the file to add it to the allow list.  Should I leave it always on, even though it logs a line each time a denied entry is opened, or should I add a log option/button for it so only people who want to see it will have it enter their logs?

File: src/main/java/codechicken/nei/NEIClientConfig.java [hunk 1]
```diff
@@ -279,10 +279,16 @@ public boolean onClick(int button) {
...
 
         tag.getTag("inventory.focusSearchWidgetOnOpen")
-                .setComment("Focus Search Widget on Open, blurs on mouse move unless typing has started first")
+                .setComment(
+                        "Focus Search Widget on Open, blurs/unfocuses on mouse move unless typing has started first")
                 .getBooleanValue(false);
         API.addOption(new OptionToggleButton("inventory.focusSearchWidgetOnOpen", true));
 
+        tag.getTag("inventory.firstInvCloseClosesInSearch").setComment(
+                "Pressing the open inventory key when the inventory\nwas just opened when the search is focused will\nclose it instead of typing in the search\n(This will prevent typing the inventory open\nkey into the search as the first input though)")
```
Comment (slprime) [inline src/main/java/codechicken/nei/NEIClientConfig.java:288]: please write a short comment

File: src/main/java/codechicken/nei/LayoutManager.java [hunk 2]
```diff
@@ -176,6 +177,7 @@ public boolean isAllowedGuiAutoSearchFocus(GuiContainer gui) {
                 return true;
             }
         }
+        NEIClientConfig.logger.info("Checking for auto focus on " + guiClassName + " had no match");
```
Comment (slprime) [inline src/main/java/codechicken/nei/LayoutManager.java:180]: It's a bad idea to spam in log because you will rarely or never configure the autofocus

Comment (OvermindDL1) [reply src/main/java/codechicken/nei/LayoutManager.java:180]: Super useful for someone making a pack though, should it be configurable or just wipe entirely and let them figure it out?

File: src/main/java/codechicken/nei/NEIClientConfig.java (hunk 1 above)
Comment (OvermindDL1) [reply src/main/java/codechicken/nei/NEIClientConfig.java]: Shortened, is this good or do you have a different wording suggestion?

**Review by slprime — APPROVED** [review]

**Comment by Caedis** [comment]
What if the first letter you want to type is the inventory key?

**Comment by OvermindDL1** [comment]
> What if the first letter you want to type is the inventory key?

Yeah that was a common thing talked about on discord.  They check around and 'e' is fairly uncommon for a first character and 'e' is default for inventory and even then it's easy to get in habit to just skip the 'e' entirely and start typing whatever's after it, and if someone did rebind they can just turn it off anyway.  The whole feature set of autofocus is off by default regardless.

**Pull request merged**